### DISCUSSION TITLE
what does this mean and how can i fix it

### DISCUSSION THREAD

**Discussion by fireblade2534** [openai/whisper#836] [Q&A]
This is the Errors: "Expected size for first two dimensions of batch2 tensor to be: [48, 249] but got: [48, 165].","Expected size for first two dimensions of batch2 tensor to be: [48, 19] but got: [48, 20]."

**Comment by ileocho** [comment]
Got this error because I was running multiple threads of recording and transcribing at the same time. This disappears when making sure to close and join your threads.

**Reply by peternasser99** [reply]
how ??

**Reply by ileocho** [reply]
I don’t how your code looks like, but check if you don’t listen to the microphone twice. If you are using an if statement to listen to your microphone, don’t forget to close it when you are done using it, otherwise you will get this error because you may have a queue with more input data than the model is supposed to process.

**Reply by peternasser99** [reply]
in my case 
iam sending requests to a django backend fast on a slow cpu which should return the transcribed text

**Reply by zainbalouch** [reply]
facing the same issue. can you help me in my code.
basically i am using websockets where audio chunks are being sent from client and it transcribe and sends back transciption.
for one client it is working fine but when i connect more than one client i am facing this error.

Error in transcription:  Expected size for first two dimensions of batch2 tensor to be: [6, 82] but got: [6, 47].
Error in transcription:  Expected size for first two dimensions of batch2 tensor to be: [6, 126] but got: [6, 82]


This is my code.

`import asyncio
import websockets
import numpy as np
import whisper
import torch
import threading
import time
from timeit import default_timer as timer

frames_lock = threading.Lock()
threshold = 50  # 50  # voice pitch in * of thousands, eg, 40 for 40000, 30 for 30000
ms = 500 # milliseconds - Default
user_data = {}
plainResponse = False
connected_clients = set()

# Load the Whisper model
model_name = "tiny.en"
print("Loading model...")
model = whisper.load_model(model_name)
print(f"Whisper Model Loaded: '{model_name}'")


async def audio_data(websocket, path):
    global connected_clients
    connected_clients.add(websocket)
    VAD = ms
    plainResponse = False
    try:
        VAD = int(VAD)
    except Exception as e:
        print(f'There was some issue in converting str(VAD) to int: {str(e)}')
        VAD = VAD
    connection_id = id(websocket)
    user_data[connection_id] = {
        'frames_in': [],
        'silence_duration': 0,
        'last_speech_time': time.time(),
        'start': False, 
        'VAD': VAD,
        'plainResponse':plainResponse
    }
    print(f'New connection: {connection_id} | Total Connected Clients: {len(connected_clients)}')
    try:
        async for message in websocket:
            user = user_data.get(connection_id)
            if not user:
                return
            audio_buffer = np.frombuffer(message, dtype=np.int16)
            b = ((np.abs(np.fft.rfft(audio_buffer))) * 16000 / len(message)//2).mean()
            if b > threshold*1000:
                user['silence_duration'] = 0
                user['start'] = True
                user['last_speech_time'] = time.time()
                user['frames_in'].append(audio_buffer)
            else:
                if user['start']:
                    user['frames_in'].append(audio_buffer)
                user['silence_duration'] = time.time() - user['last_speech_time']
            if user['silence_duration'] * 1000 >=  user['VAD'] and user['start']:
                user['start'] = False
                if user['frames_in']:
                    with frames_lock:
                        transcription_thread = threading.Thread(target=transcribe, args=(user['frames_in'].copy(), user['VAD'], user['plainResponse'], connection_id, websocket))                    
                        transcription_thread.daemon = True
                        transcription_thread.start()
                        user['frames_in'].clear()  
            await websocket.send(f"userIsSilent")
    finally:
        # Cleanup after the client disconnects
        connected_clients.remove(websocket)
        print(f'Removed connection: {connection_id} | Total Connected Clients: {len(connected_clients)}')
        return True

def transcribe(frames, vad, plainResponse, room_connection_id, sio):
    asyncio.run(transcribe_async(frames, vad, plainResponse, room_connection_id, sio))

async def transcribe_async(frames, vad, plainResponse, room_connection_id, websocket):
    try:
            start_time    = timer()
            audio_data_np = np.frombuffer(b''.join(frames), np.int16).flatten().astype(np.float32) / 32768.0
            audio_tensor = torch.from_numpy(audio_data_np)
            transcription = model.transcribe(audio_tensor)
            end_time      = timer()
            elapsed  = (end_time - start_time)
            print(f"Sockets Connected: {len(connected_clients)} | Time: {elapsed} | {len(audio_data_np)}| {room_connection_id} --> {transcription['text']}")
            await websocket.send(transcription["text"])

    except Exception as e:
        print('Error in transcription: ', str(e))

# Start the WebSocket server
start_server = websockets.serve(audio_data, "localhost", 8765)

# Run the server until stopped
asyncio.get_event_loop().run_until_complete(start_server)
asyncio.get_event_loop().run_forever()
`
